modules/login/php/endpoints/authentication.class.inc

```diff
@@ -0,0 +1,453 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used to retrieve info with authentication by the login form.
+ *
+ * PHP Version 7
+ *
+ * @category LORIS_Module
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\login\endpoints;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+use \LORIS\api\Endpoint;
+use SinglePointLogin;
+use State;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Authentication extends Endpoint implements ETagCalculator
+{
+    public $skipTemplate = true;
+
+    /**
+     * This function will return a json object for login.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        case 'POST':
+            return $this->_handlePOST($request);
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
```

Comment:
Each of password reset, request account, and login make more sense as POST requests than GET requests.